src/Features/VisualBasic/Completion/VisualBasicCompletionRules.vb

```diff
@@ -14,6 +14,38 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion
                 MyBase.New(service)
             End Sub
 
+            Protected Overrides Function CompareMatches(leftMatch As PatternMatch, rightMatch As PatternMatch, leftItem As CompletionItem, rightItem As CompletionItem) As Integer
+                Dim diff As Integer
+                diff = PatternMatch.CompareType(leftMatch, rightMatch)
+                If diff <> 0 Then
+                    Return diff
+                End If
+
+                diff = PatternMatch.CompareCamelCase(leftMatch, rightMatch)
+                If diff <> 0 Then
+                    Return diff
+                End If
+
+                ' More important than the case sensitivity is that "left" isn't an argument name
+                Dim leftIsNamedArgument = TypeOf leftItem.CompletionProvider Is NamedParameterCompletionProvider
+                Dim rightIsNamedArgument = TypeOf rightItem.CompletionProvider Is NamedParameterCompletionProvider
+                If leftIsNamedArgument AndAlso Not rightIsNamedArgument Then
```

Comment:
Does this comparer have to be reflexive? I could imagine we might invoke very strange sorting bugs since this function is not.